catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java

```diff
@@ -41,23 +42,30 @@
         Function<T, Serializable> accessor = attribute.getAccessorFunction();
         Serializable value = accessor.apply(segment);
 
-        AttributeDescriptor descriptor = attribute.getAttributeDescriptor();
+        Set<AttributeDescriptor> descriptors = attribute.getAttributeDescriptors();
 
-        if (descriptor == null) {
+        if (descriptors == null) {
             LOGGER.debug("Could not set metacard attribute " + attribute.getLongName()
                     + " since it does not belong to this metacard type");
             return;
         }
 
-        if (descriptor.getType().equals(BasicTypes.STRING_TYPE) && value != null
-                && ((String) value).length() == 0) {
-            value = null;
+        for (AttributeDescriptor descriptor : descriptors) {
+            if (descriptor.getType()
+                    .equals(BasicTypes.STRING_TYPE) && value != null
+                    && value.toString().length() == 0) {
+                value = null;
+            }
         }
 
-        if (value != null) {
-            Attribute catalogAttribute = populateAttribute(metacard, descriptor.getName(), value);
-            LOGGER.trace("Setting the metacard attribute [{}, {}]", descriptor.getName(), value);
-            metacard.setAttribute(catalogAttribute);
+        for (AttributeDescriptor descriptor : descriptors) {
+            if (value != null) {
+                Attribute catalogAttribute = populateAttribute(metacard, descriptor.getName(), value);
+                LOGGER.trace("Setting the metacard attribute [{}, {}]",
```

Comment:
I think that according to the trace description on our Log Levels this seems appropriate since it is tracking program execution.